File: java/pyl/33996/code/src/main/java/io/temporal/poc/sample/SampleStarter.java
package io.temporal.poc.sample;

import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowOptions;
import io.temporal.serviceclient.WorkflowServiceStubs;

import java.util.UUID;

public class SampleStarter {

    public static void main(String[] args) {
        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
        WorkflowClient client = WorkflowClient.newInstance(service);




        GreetingWorkflow workflow = client.newWorkflowStub(
                GreetingWorkflow.class,
                WorkflowOptions.newBuilder()
                        .setTaskQueue(SampleWorker.TASK_QUEUE)
                        .setWorkflowId("sample-" + UUID.randomUUID())
                        .build());

        WorkflowClient.start(workflow::greet, "");



        try {
            Thread.sleep(2_000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }

        workflow.signal("test");





        try {
            Thread.sleep(40_000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }

        System.out.println("Result: " );
    }
}

File: java/pyl/33996/code/kotlin-2.2.0/src/main/kotlin/io/temporal/poc/sample/SampleStarter.kt
package io.temporal.poc.sample

import io.temporal.client.WorkflowClient
import io.temporal.client.WorkflowOptions
import io.temporal.serviceclient.WorkflowServiceStubs


object SampleStarter {
    @JvmStatic
    fun main(args: Array<String>) {
        execute()
    }

    fun execute(): String {
        val service = WorkflowServiceStubs.newLocalServiceStubs()
        val client = WorkflowClient.newInstance(service)


        val workflow = client.newWorkflowStub(
            GreetingWorkflow::class.java,
            WorkflowOptions.newBuilder()
                .setTaskQueue(SampleWorker.TASK_QUEUE)
                .setWorkflowId(WorkflowId.value)
                .build()
        )

        val result = workflow.greet("world")
        println("Result: $result")

        val historyAsJson = client.fetchHistory(WorkflowId.value).toJson(true)

        return historyAsJson
    }
}
